File: app/src/main/java/i/am/shiro/amai/fragment/ReadFragment.java
package i.am.shiro.amai.fragment;

import android.content.Context;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import java.util.List;

import i.am.shiro.amai.R;
import i.am.shiro.amai.adapter.BookPageAdapter;
import i.am.shiro.amai.model.Book;
import i.am.shiro.amai.model.Image;
import i.am.shiro.amai.widget.PageRecyclerView;
import io.realm.Realm;

import static android.view.WindowManager.LayoutParams.FLAG_FULLSCREEN;
import static androidx.core.view.ViewCompat.requireViewById;

public class ReadFragment extends Fragment {

    private static final String BOOK_ID = "bookId";

    private static final String PAGE_INDEX = "pageIndex";

    private Realm realm;

    public ReadFragment() {
        super(R.layout.fragment_read);
    }

    @NonNull
    static Fragment newInstance(Book book, int pageIndex) {
        Bundle args = new Bundle();
        args.putInt(BOOK_ID, book.getId());
        args.putInt(PAGE_INDEX, pageIndex);

        Fragment fragment = new ReadFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);

        realm = Realm.getDefaultInstance();

        requireActivity()
            .getWindow()
            .addFlags(FLAG_FULLSCREEN);
    }

    @Override
    public void onDetach() {
        super.onDetach();

        realm.close();

        requireActivity()
            .getWindow()
            .clearFlags(FLAG_FULLSCREEN);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        BookPageAdapter adapter = new BookPageAdapter(this, extractImages());

        PageRecyclerView pageRecycler = requireViewById(view, R.id.pageRecycler);
        pageRecycler.setHasFixedSize(true);
        pageRecycler.setAdapter(adapter);
        pageRecycler.requestFocus();

        if (savedInstanceState == null) {
            pageRecycler.scrollToPosition(extractPageIndex());
        }
    }

    private List<Image> extractImages() {
        int bookId = getArguments().getInt(BOOK_ID, -1);
        return realm.where(Book.class)
            .equalTo("id", bookId)
            .findFirst()
            .getPageImages();
    }

    private int extractPageIndex() {
        return getArguments()
            .getInt(PAGE_INDEX, 0);
    }
}

File: app/src/main/java/i/am/shiro/amai/fragment/ReadFragment.kt
package i.am.shiro.amai.fragment

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.compose.ui.platform.ComposeView
import androidx.compose.ui.platform.ViewCompositionStrategy
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat
import androidx.fragment.app.Fragment
import i.am.shiro.amai.compose.AmaiTheme
import i.am.shiro.amai.compose.ReadScreen
import i.am.shiro.amai.util.amaiViewModels
import i.am.shiro.amai.util.argument
import i.am.shiro.amai.viewmodel.ReadViewModel

class ReadFragment() : Fragment() {

    private val viewModel by amaiViewModels<ReadViewModel>()

    private var bookId by argument<Int>()

    private var pageIndex by argument<Int>()

    constructor(bookId: Int, pageIndex: Int) : this() {
        this.bookId = bookId
        this.pageIndex = pageIndex
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)

        val window = requireActivity().window
        WindowInsetsControllerCompat(window, window.decorView).apply {
            hide(WindowInsetsCompat.Type.statusBars())
            systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        }
    }

    override fun onDetach() {
        super.onDetach()

        val window = requireActivity().window
        WindowInsetsControllerCompat(window, window.decorView).show(WindowInsetsCompat.Type.statusBars())
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        viewModel.setBookId(bookId)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return ComposeView(requireContext()).apply {
            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
            setContent {
                AmaiTheme {
                    ReadScreen(
                        viewModel = viewModel,
                        initialPage = pageIndex
                    )
                }
            }
        }
    }
}
